modules/services/git-workspace.nix

```
@@ -0,0 +1,153 @@
+            </para><para>
+        It contains lines like
+        <literal>
+        GITHUB_TOKEN=ghp_...
+        </literal>
+            </para><para>
+        If you want to use Nix to manage these tokens use a tool like sops-nix
+        or agenix. This file must be readable by the home-manager user.
+      '';
+      example = "/var/lib/git-workspace-tokens";
+    };
+    workspaces = lib.mkOption {
+      type = lib.types.attrsOf workspaceType;
+      default = { };
+      description = ''
+        1:1 representation of a single <literal>workspace.toml</literal> consumed
+        by git-workspace.
+      '';
+      example = lib.literalExpression ''
+        {
+          personal.provider = [
+            {
+              provider = "github"
+              name = "github_username"
+              path = "''${config.home.homeDirectory}/personal_projects"
+              skip_forks = false;
+            }
+            {
+              provider = "gitlab"
+              name = "gitlab_username"
+              # it's possible to use the same path as previous provider
+              # if there are not name clashes
+              path = "''${config.home.homeDirectory}/personal_projects"
+            }
+          ];
+          work.provider = [
+            {
+              provider = "github"
+              name = "github_org"
+              path = "''${config.home.homeDirectory}/work"
+            }
+          ];
+        };
+      '';
+    };
+  };
+  config = lib.mkIf cfg.enable {
+    assertions = [
+      (lib.hm.assertions.assertPlatform "services.git-workspace" pkgs
+        lib.platforms.linux)
+    ];
+    xdg.configFile = lib.mapAttrs' (workspaceName: workspace: {
+      name = "git-workspace/${workspaceName}/workspace.toml";
+      value.source = (tomlFormat.generate "${workspaceName}-workspace.toml"
+        workspace).outPath;
+    }) cfg.workspaces;
+    systemd.user.services = lib.mapAttrs' (workspaceName: workspace: rec {
+      name = "git-workspace-${workspaceName}-update";
+      value = {
+        Unit.Description = "Runs `git-workspace update` for ${workspaceName}";
+        Service = {
+          EnvironmentFile = cfg.environmentFile;
+          ExecStart = let
+            script = pkgs.writeShellApplication {
+              name = "${name}-launcher";
+              text = ''
+                ${cfg.package}/bin/git-workspace \
+                  --workspace ${config.xdg.configHome}/git-workspace/${workspaceName} \
+                  update
+              '';
+              runtimeInputs = with pkgs; [ busybox openssh git ];
+            };
+          in "${script}/bin/${name}-launcher";
+        };
+      };
+    }) cfg.workspaces;
+    systemd.user.timers = lib.mapAttrs' (workspaceName: workspace: {
+      name = "git-workspace-${workspaceName}-update";
+      value = {
+        Unit = {
+          Description =
+            "Automatically runs `git-workspace update` for ${workspaceName}";
```

Can simplify here as well
```suggestion
            "git-workspace update for ${workspaceName}";
```